op-bindings/bindgen/generator_remote.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ethereum-optimism/optimism/op-bindings/etherscan"
+)
+
+type bindGenGeneratorRemote struct {
+	bindGenGeneratorBase
+	contractDataClients map[string]contractDataClient
+	tempArtifactsDir    string
+}
+
+type contractDataClient interface {
+	FetchAbi(address string) (string, error)
+	FetchDeployedBytecode(address string) (string, error)
+	FetchDeploymentTxHash(address string) (string, error)
+	FetchDeploymentTx(txHash string) (etherscan.TxInfo, error)
+}
+
+type remoteContract struct {
+	Name            string            `json:"name"`
+	Verified        bool              `json:"verified"`
+	Deployments     map[string]string `json:"deployments"`
```

Comment:
Couldn't `Deployments` be a struct since the keys are known upfront? Then the values could be `common.Address`